pilot/pkg/model/config.go

```diff
@@ -45,6 +45,40 @@ const (
 	RevisionLabel = "istio.io/rev"
 )
 
+// ConfigKey describe a specific config item depends on or get changed.
+// In most cases name is config name except that for ServiceEntry it's Service.Hostname.
+type ConfigKey struct {
+	Kind      resource.GroupVersionKind
+	Name      string
+	Namespace string
+}
+
+// ConfigsOfKind extracts configs of the specified kind.
+func ConfigsOfKind(configs map[ConfigKey]struct{}, kind resource.GroupVersionKind) map[ConfigKey]struct{} {
+	ret := make(map[ConfigKey]struct{})
+
+	for conf := range configs {
+		if conf.Kind == kind {
+			ret[conf] = struct{}{}
+		}
+	}
+
+	return ret
+}
+
+// ConfigNamesOfKind extracts config names of the specified kind.
+func ConfigNamesOfKind(configs map[ConfigKey]struct{}, kind resource.GroupVersionKind) map[string]struct{} {
+	ret := make(map[string]struct{})
+
+	for conf := range configs {
+		if conf.Kind == kind {
+			ret[conf.Name] = struct{}{}
+		}
+	}
+
+	return ret
+}
```

Comment:
This is not right for other configs as the namespace is not respected. 
Return `map[struct{name, namespace}]`

Reply:
I know that. But in some cases we just need name itself. Such as the legacy `edsUpdates`.

Reply:
Do you actually need this function? Can you use `ConfigsOfKind` and derive config names wherever needed like legacy `edsUpdates` for example?

Reply:
It's a util function to avoid repeat the "derive config names" code here and there.